### PR TITLE
Partial (2/2) spanish translation ch10-01-syntax.md

### PR DESCRIPTION
**Pull request by jossue33**
Number of words: 1094
Line: 261-476

### CODE REVIEW

File: second-edition/src/ch10-01-syntax.md [hunk 1]
```diff
@@ -258,12 +258,12 @@ hard to read and understand. If you get to a point of needing lots of generic
 types, it’s probably a sign that your code could use some restructuring to be
 separated into smaller pieces.
 
-### Using Generic Data Types in Enum Definitions
+### Usando tipos de datos genéricos en definiciones Enum
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: Again, please write titles with important words capitalized

Usando Tipos de Datos...

File: second-edition/src/ch10-01-syntax.md [hunk 2]
```diff
@@ -272,16 +272,16 @@ enum Option<T> {
...
 ```
 
-In other words, `Option<T>` is an enum generic in type `T`. It has two
-variants: `Some`, which holds one value of type `T`, and a `None` variant that
-doesn’t hold any value. The standard library only has to have this one
-definition to support the creation of values of this enum that have any
-concrete type. The idea of “an optional value” is a more abstract concept than
-one specific type, and Rust lets us express this abstract concept without lots
-of duplication.
+En otras palabras, `Option<T>` es un genérico en tipo `T`. Tiene dos
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: es un Enum genérico de tipo

File: second-edition/src/ch10-01-syntax.md [hunk 3]
```diff
@@ -272,16 +272,16 @@ enum Option<T> {
...
+variantes: `Some`, que alberga un valor de tipo `T`, y una variante `None` que 
+no alberga ningún valor. La biblioteca estandar solo debe de tener esta única
+definición para respaldar la creación de valores en este enum que tenga cualquier
+tipo concreto. La idea de “un valor opcional” es más un concepto abstracto que
+un tipo específico, y Rust nos deja expresar este concepto abstracto sin muchas
+duplicaciones.
 
-Enums can use multiple generic types as well. The definition of the `Result`
-enum that we used in Chapter 9 is one example:
+Enums también pueden usar multiples tipos genéricos. La definición de el enum `Result`
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: del

File: second-edition/src/ch10-01-syntax.md [hunk 4]
```diff
@@ -290,28 +290,28 @@ enum Result<T, E> {
...
-definitions that differ only in the types of the values they hold, you can
-remove the duplication by using the same process we used with the function
-definitions to introduce generic types instead.
+Cuando reconoces situaciones en tu código con multiple estructura o definiciones
+enum que eran diferentes solo en los tipos de los valores que albergaban, puedes
+retirar la duplicación al usar el mismo proceso que usamos con las definiciones de
+función para introducir tipos genéricos en su lugar.
 
-### Using Generic Data Types in Method Definitions
+### Usando tipos de datos genericos en definiciones de metodos
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: Capitalize please

File: second-edition/src/ch10-01-syntax.md [hunk 5]
```diff
@@ -290,28 +290,28 @@ enum Result<T, E> {
...
 
-Like we did in Chapter 5, we can implement methods on structs and enums that
-have generic types in their definitions. Listing 10-9 shows the `Point<T>`
-struct we defined in Listing 10-6. We’ve then defined a method named `x` on
-`Point<T>` that returns a reference to the data in the field `x`:
+Como hicimos en el capitulo 5, podemos implementar metodos en estructuras y enums que
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: Theres no punctuation for _método_ here

File: second-edition/src/ch10-01-syntax.md [hunk 6]
```diff
@@ -357,26 +357,26 @@ impl Point<f32> {
...
-types than the `self` `Point` that we’re calling `mixup` on. The method creates
-a new `Point` instance that has the `x` value from the `self` `Point` (which is
-of type `T`) and the `y` value from the passed-in `Point` (which is of type
-`W`):
+<span class="caption">Listado 10-10: Construyendo un bloqueo `impl`que solo 
+aplica a una estructura con un tipo especifico es usada paea el parametro de tipo generico
+ `T`</span>
 
-<span class="filename">Filename: src/main.rs</span>
+Este codigo significa que el tipo `Point<f32>` tendrá un metodo llamado
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: Again, no punctuation for metodo or parametro

File: second-edition/src/ch10-01-syntax.md [hunk 7]
```diff
@@ -403,54 +403,54 @@ fn main() {
 }
 ```
 
-<span class="caption">Listing 10-11: Methods that use different generic types
-than their struct’s definition</span>
+<span class="caption">Listado 10-11: Metodos que usan diferentes tipos genericos
+que su definición de estructura</span>
+
+En `main`, hemos definido un `Point` que tiene un `i32` por `x` (con valor `5`)
+y un `f64` por `y` (con valor `10.4`). `p2` es un `Point` que tiene un pedazo de cordón de `x` (con valor `"Hello"`) y un `char` por `y` (con valor `c`).
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: pedazo de cordon -> trozo de cadena

File: second-edition/src/ch10-01-syntax.md [hunk 8]
```diff
@@ -403,54 +403,54 @@ fn main() {
...
+Te estarás preguntando mientras leías esta sección si hay un tiempo de coste en el tiempo de ejecución
+al usar tipos de parámetros genéricos. Tengo buenas noticias: ¡La forma en la que Rust ha implementado los
+genéricos significa que tu código no se ejecutará más lento como si hubieses especificado
+tipos concretos en vez de tipos de parámetros genéricos!
 
-You may have been reading this section and wondering if there’s a run-time cost
-to using generic type parameters. Good news: the way that Rust has implemented
-generics means that your code will not run any slower than if you had specified
-concrete types instead of generic type parameters!
+Rust logra esto al realizar una  *monotransformación* del codigo usando los genéricos
```
Comment (thecodix) [inline second-edition/src/ch10-01-syntax.md]: codigo has no punctuaction, also _generico_ not enerico

**Comment by thecodix** [comment]
Please review use of spanish accents here

**Comment by jossue33** [comment]
@thecodix todo listo.

**Comment by thecodix** [comment]
machine translated

**Pull request merged**